src/library.js

```diff
@@ -657,7 +657,23 @@ addToLibrary({
 
     {{{ makeSetValue('daylight', '0', 'Number(winterOffset != summerOffset)', 'i32') }}};
 
-    var extractZone = (date) => date.toLocaleTimeString(undefined, {hour12:false, timeZoneName:'short'}).split(' ')[1];
+    var extractZone = (date) => {
+      var timezoneOffset = date.getTimezoneOffset();
+      if (isNaN(timezoneOffset) || timezoneOffset === 0) {
```

Comment:
Can timezoneOffset actually be NaN here?     It sees like both `winter` and `summer` but be valid dates (if you like we can assert that above, but I don't think its necessary).

Reply:
Looking at the code you are right, it can't be an invalid date, removed the extra check in https://github.com/emscripten-core/emscripten/pull/22250/commits/f3eaebfdb3381b49aae286461685e804a64f2268